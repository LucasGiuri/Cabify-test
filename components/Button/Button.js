import './Button.scss';
import PropTypes from 'prop-types';

const Button = ({onClick, text, classes, isDisabled}) => {
  let buttonClasses = ['button'];

  if (classes) {
    buttonClasses = [...buttonClasses, classes];
  }

  return (
    <button onClick={onClick} className={buttonClasses.join(' ')} disabled={isDisabled}>{text}</button>
  )
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
  isDisabled: PropTypes.bool,
  classes: PropTypes.array
};

Button.defaultProps = {
  onClick: () => '',
  classes: [''],
  isDisabled: false
};

export default Button;