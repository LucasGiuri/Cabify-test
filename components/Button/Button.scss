@import '../../scss/colors';
@import '../../scss/breakpoints';

.button {
  width: 100%;
  padding: 6px 12px;
  border: 2px solid $black;
  background: $white;
  font-weight: bold;
  cursor: pointer;
  
  &.button-primary {
    border-color: $green;
    color: $green;

    &:hover {
      background: $green;
      color: $white;
    }
  }

  &.button-danger {
    border-color: $red;
    color: $red;
    
    &:hover {
      background: $red;
      color: $white;
    }
  }

  &:disabled {
    border-color: $grey;
    color: $grey;

    &:hover {
      background: $white;
      color: $grey;
      cursor: not-allowed;
    }
  }

  &:hover {
    background: $black;
    color: $white;
  }
}

@media #{breakpoint('medium')} {
  .button {
    max-width: 12em;
  }
}