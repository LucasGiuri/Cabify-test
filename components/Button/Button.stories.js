import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Button')
  .add('Button default', () => (
    <Button text="FakeText"/>
  ))
  .add('Button with class button-primary', () => (
    <Button classes={['button-primary']} text="FakeText"/>
  ))
  .add('Button with class button-danger', () => (
    <Button classes={['button-danger']} text="FakeText"/>
  ))
  .add('Button with isDisabled=true', () => (
    <Button isDisabled={true} text="FakeText"/>
  ))