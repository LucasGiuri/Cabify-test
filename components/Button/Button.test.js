import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';

describe('Button', () => {
  it('Button recives required props. matchToSnapshot', () => {
    const text = 'FakeText';
    const component = renderer.create(<Button text={text} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Button with isDisabled=true. matchToSnapshot', () => {
    const text = 'FakeText';
    const component = renderer.create(<Button text={text} isDisabled={true}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Button with class button-primary. matchToSnapshot', () => {
    const text = 'FakeText';
    const component = renderer.create(<Button text={text} isDisabled={true} classes={['button-primary']} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Button with class button-danger. matchToSnapshot', () => {
    const text = 'FakeText';
    const component = renderer.create(<Button text={text} isDisabled={true} classes={['button-danger']} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Button prop onClick is triggered. onClick prop should be called as expected', () => {
    const onClick = jest.fn();
    const text = 'FakeText';
    const componentToTest = (
      <Button text={text} onClick={onClick} />
    );
    const tree = renderer.create(componentToTest);
    const button = tree.root.findByType(Button);
    button.props.onClick();
    expect(onClick).toHaveBeenCalledTimes(1);
    expect(onClick).toHaveBeenCalledWith();
  })
})