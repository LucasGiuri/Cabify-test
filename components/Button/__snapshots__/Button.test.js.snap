// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button recives required props. matchToSnapshot 1`] = `
<button
  className="button "
  disabled={false}
  onClick={[Function]}
>
  FakeText
</button>
`;

exports[`Button Button with class button-danger. matchToSnapshot 1`] = `
<button
  className="button button-danger"
  disabled={true}
  onClick={[Function]}
>
  FakeText
</button>
`;

exports[`Button Button with class button-primary. matchToSnapshot 1`] = `
<button
  className="button button-primary"
  disabled={true}
  onClick={[Function]}
>
  FakeText
</button>
`;

exports[`Button Button with isDisabled=true. matchToSnapshot 1`] = `
<button
  className="button "
  disabled={true}
  onClick={[Function]}
>
  FakeText
</button>
`;
